app/src/main/kotlin/at/bitfire/davdroid/repository/DavCollectionRepository.kt

```diff
@@ -343,9 +347,16 @@ class DavCollectionRepository @Inject constructor(
                                 text(DavUtils.ARGBtoCalDAVColor(it))
                             }
                         }
-                        timezoneDef?.let {
-                            insertTag(CalendarTimezone.NAME) {
-                                cdsect(it)
+                        timezoneDef?.let { timezoneId ->
+                            insertTag(CalendarTimezoneId.NAME) {
+                                text(timezoneId)
+                            }
+                            getVTimeZone(timezoneId)?.let {
+                                insertTag(CalendarTimezone.NAME) {
+                                    text(
+                                        Calendar(ComponentList(listOf(it))).toString()
```

Comment:
Also maybe a comment like `// spec requires "an iCalendar object with exactly one VTIMEZONE component"` to see what this is for.